solidity/contracts/deposit/DepositUtils.sol

```diff
@@ -458,56 +464,70 @@ library DepositUtils {
+        uint256 signerFeeAdjust = computeSignerFeeAdjustment(_d, redeemerPaysNoFee, frtExists);
+        return mainCharge.add(frtAdjust).add(signerFeeAdjust);
+    }
 
-        if (ownerRedemption) {
+    /// @notice                    Get the base TBTC amount needed to redeem.
+    /// @param tdtOwnerRedemption  True if the redeemer is the TDT holder.
+    /// @return                    The amount in TBTC.
+    function computeBaseRedemptionCharge(
+        DepositUtils.Deposit storage _d,
+        bool tdtOwnerRedemption
+    ) internal view returns (uint256){
+        if (tdtOwnerRedemption) {
             return 0;
         }
-
         return lotSizeTbtc(_d);
     }
 
-    /// @notice             Get the fee rebate amount needed to redeem.
-    /// @param _redeemer    The deposit redeemer.
-    /// @return             The amount in TBTC.
-    function frtAdjust(DepositUtils.Deposit storage _d, address _redeemer) internal view returns (uint256){
+    /// @notice                    Get the fee rebate amount needed to redeem.
+    /// @param _redeemerPaysNoFee  True if Deposit is at term or in courtesy call.
+    /// @param _frtOwnerRedemption True if the redeemer is the FRT holder.
+    /// @param _frtExists          True if the FRT exists.
+    /// @return                    The amount in TBTC.
+    function computeFrtAdjustment(
+        DepositUtils.Deposit storage _d,
+        bool _redeemerPaysNoFee,
+        bool _frtOwnerRedemption,
+        bool _frtExists
+    ) internal view returns (uint256){
         if(
-            _d.inCourtesyCall() ||
-            remainingTerm(_d) == 0 ||
-            feeRebateTokenHolder(_d) == address(0) ||
-            feeRebateTokenHolder(_d) == _redeemer
+            _redeemerPaysNoFee ||
+            _frtOwnerRedemption ||
+            !_frtExists
         ){
             return 0;
         }
 
         return signerFee(_d);
     }
 
-    /// @notice             Get the signer fee amount needed to redeem.
-    /// @param _redeemer    The deposit redeemer.
-    /// @return             The amount in TBTC.
-    function signerFeeAdjust(DepositUtils.Deposit storage _d, address _redeemer) internal view returns (uint256){
-        if(
-            remainingTerm(_d) == 0 ||
-            _d.inCourtesyCall()
-        ){
+    /// @notice                   Get the signer fee amount needed to redeem.
+    /// @param _redeemerPaysNoFee True if Deposit is at term or in courtesy call.
+    /// @param _frtExists         True if the FRT exists.
+    /// @return                   The amount in TBTC.
+    function computeSignerFeeAdjustment(
+        DepositUtils.Deposit storage _d,
+        bool _redeemerPaysNoFee,
+        bool _frtExists
+    ) internal view returns (uint256){
+        if(_redeemerPaysNoFee){
             return 0;
         }
-        if(feeRebateTokenHolder(_d) != address(0)){
+        if(_frtExists){
```

Comment:
We should settle on either the structure in `computeFrtAdjustment` (single if that returns 0) or the one here (multiple ifs that all return 0), but let's stick to one or the other.